worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionDataWriter.java

```diff
@@ -368,23 +368,22 @@ protected void waitOnNoPending(AtomicInteger counter) throws IOException {
   }
 
   protected void takeBuffer() {
-    // metrics start
-    String metricsName = null;
-    String fileAbsPath = null;
     if (source.metricsCollectCriticalEnabled()) {
+      // metrics start
+      String metricsName = null;
```

Comment:
```suggestion
      String metricsName = WorkerSource.TAKE_BUFFER_TIME();
```
Meanwhile, remove line 375.

Reply:
Done